cloudinit/sources/DataSourceCloudStack.py

```diff
@@ -282,7 +289,16 @@ def get_vr_address():
         return latest_address
 
     # Try dhcp lease files next
-    lease_file = dhcp.IscDhclient.get_latest_lease()
+    # get_latest_lease() needs a Distro object to know which directory
+    # stores lease files
+    lease_file = None
+    if distro:
+        lease_file = dhcp.IscDhclient.get_latest_lease(
+            distro.dhclient_lease_directory, distro.dhclient_lease_file_regex
+        )
+    else:
+        LOG.debug("Distro object is not defined, skipping leasefile search")
```

Comment:
and if it did, i don't see how a debug log is useful.

remove the if, and the else, let it crash loudly and clearly

Reply:
Updated the PR: removed the distro=None checks, as suggested.